app/views/admin/pre_auth_states/index.html.erb

```diff
@@ -41,6 +45,30 @@
         </div>
       <% end %>
     </div>
+  <% end.empty? %>
```

Comment:
Haven't reviewed the whole thing completely yet but this line does not look correct to me because `each` doesn't return anything like the way `map` does, so `empty?` will probably always evaluate to `true`. The fix _might be_ simply replacing `each` with `map`. Even better, though, would be to do `if`...`else`, more Ruby-like.

Reply:
I tested this.. it works

Reply:
Ok, I'm surprised (it ain't Ruby-like) but I believe you; might be worth writing a feature test though?